### PR TITLE
Remove package:typed_data from package:flutter dependencies

### PR DESCRIPTION
**Pull request by jonahwilliams**
Replace the usaged of the Uint8Buffer from package:typed_data with a List<int>. This actually improves performance on wembly, using the following benchmark:

```dart
import 'package:flutter/services.dart';

ByteData? result;

void main() {
  for (var x = 0; x < 10; x++) {
    var codec = StandardMessageCodec();
    var sw = Stopwatch()..start();
    for (var i = 0; i < 100; i++) {
      result = codec.encodeMessage({
        for (var i = 0; i < 200; i++)
          i: [for (var j = 0; j < 23; j++) j]
      });
    }
    print(sw.elapsedMilliseconds);
    print(result);
  }
}
```

ToT:
```
An Observatory debugger and profiler on wembley is available at: http://127.0.0.1:61792/wKRO2rVZZf4=/
I/flutter ( 4652): 488
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 451
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 449
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 449
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 447
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 448
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 449
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 449
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 450
I/flutter ( 4652): TypedDataView(cid: 148)
I/flutter ( 4652): 448
I/flutter ( 4652): TypedDataView(cid: 148)
```

With Change
```
An Observatory debugger and profiler on wembley is available at: http://127.0.0.1:61886/wBKsj-QFjxY=/
I/flutter ( 4807): 456
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 372
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 363
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 368
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 364
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 362
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 363
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 363
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 363
I/flutter ( 4807): TypedDataView(cid: 148)
I/flutter ( 4807): 364
I/flutter ( 4807): TypedDataView(cid: 148)
```

### CODE REVIEW

File: packages/flutter/lib/src/foundation/serialization.dart
Comment (jonahwilliams) [inline packages/flutter/lib/src/foundation/serialization.dart:15]: Allows avoiding a `late` for `_eightBytesAsList`

File: packages/flutter/pubspec.yaml
Comment (jonahwilliams) [inline packages/flutter/pubspec.yaml:14]: Need to run update-packages, delaying until approved due to churn

**Review by dnfield — APPROVED** [review]
Lgtm as long as update packages is happy 

I wonder how much we're saving by avoiding the late field vs the list.

**Comment by dnfield** [comment]
Do we have some list of forbidden packages too?  If so we should add this one.

**Comment by jonahwilliams** [comment]
Probably not much but IMO late is best avoided if its easy to

**Comment by dnfield** [comment]
Late requires extra runtime checks on every access

**Comment by jonahwilliams** [comment]
Yup, and I'm sure it makes some difference. Though I prefer to avoid it not for performance reasons but because it more or less disables the static analysis you get for correct initialization.

**Comment by jonahwilliams** [comment]
Turns out package:typed_data is still used elsewhere. Will follow up to see if it is easy to remove

**Comment by jonahwilliams** [comment]
Requires a g3fix to update BUILD file

**Comment by jonahwilliams** [comment]
Looks like flutter_goldens_client uses package:crypto for md5 encoding which pulls in package:typed_data

File: dev/integration_tests/flutter_gallery/linux/flutter/generated_plugins.cmake
Comment (goderbauer) [inline dev/integration_tests/flutter_gallery/linux/flutter/generated_plugins.cmake:9]: Is the change to this file related to this PR?

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/integration_tests/flutter_gallery/linux/flutter/generated_plugins.cmake
Comment (dnfield) [reply dev/integration_tests/flutter_gallery/linux/flutter/generated_plugins.cmake:9]: This probably comes from running pub get aftr the FFI plugin change.

Comment (jonahwilliams) [reply dev/integration_tests/flutter_gallery/linux/flutter/generated_plugins.cmake:9]: This gets auto updated whenever I run it, and it looks like its not gitignored....

Should be safe to check in :D

**Pull request merged**
